### PR TITLE
Add validation for num_realizations > 1 for analysis modules

### PR DESCRIPTION
**Pull request by xjules**
**Issue**
Resolves https://github.com/equinor/ert/issues/8348

~~blocked by semeio tests~~ should be ok now

**Approach**
Inject validation where needed.

(Screenshot of new behavior in GUI if applicable)

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by berland** [comment]
rebase to trigger rerun of semeio?

**Comment by berland** [comment]
Could there be something in the user documentation that needs to be updated following this change?

**Comment by xjules** [comment]
> Could there be something in the user documentation that needs to be updated following this change?

I thought it is somewhat implicit ert knowledge. But you are right, I'll add it.

File: src/ert/run_models/model_factory.py [hunk 1]
```diff
@@ -227,20 +237,18 @@ def _setup_multiple_data_assimilation(
...
+            active_realizations=active_realizations.tolist(),
             target_ensemble=_iterative_ensemble_format(config, args),
             weights=args.weights,
             restart_run=restart_run,
             prior_ensemble_id=prior_ensemble,
-            starting_iteration=storage.get_ensemble(prior_ensemble).iteration + 1
-            if restart_run
-            else 0,
+            starting_iteration=(
+                storage.get_ensemble(prior_ensemble).iteration + 1 if restart_run else 0
```
Comment (oyvindeide) [inline src/ert/run_models/model_factory.py:250]: Just cosmetic? No problem for me, but suggest putting in a separate commit.

**Review by oyvindeide — APPROVED** [review]
LGTM!

**Pull request merged**